### ISSUE TITLE
Exit status 106 on Windows 10 installation, with Python from Windows Store

### ISSUE THREAD

**Issue by fenekku** [lektor/lektor#798]
Windows installation isn't working for colleague and I.

**Steps to reproduce**:
1- Open up a cmd.exe command prompt as administrator
2- Copy and paste per [instructions](https://www.getlektor.com/downloads/), the following snippet: `@powershell -NoProfile -Command "(new-object net.webclient).DownloadString('https://www.getlektor.com/installer.py') | python"
`
Expect:
3- Lektor to get installed!
Get: (with <path to python.exe> being shorthand)
3- `Error: Command '[C:\\'<path to python.exe>', '-Im', 'ensurepip', '--upgrade', '--default-pip']'` returned non-zero exit status 106.

**Environment**:
- Windows 10 Pro 1909
- Python 3.8.4
- pip 20.1.1
- imagemagick installed via chocolatey

Thanks for letting us know what we are missing here!

**Comment by xlotlu** [comment]
> 3- `Error: Command '[C:\\'<path to python.exe>', '-Im', 'ensurepip', '--upgrade', '--default-pip']'` returned non-zero exit status 106.

@fenekku please do paste the `'<path to python.exe>'`, edit any sensitive data if needed.

How is your Python installed? Official installer? Windows store? Something else?

**Comment by fenekku** [comment]
Sorry about the delay. My colleague was finally able to make it work on her computer (python 3.6). Still doesn't work on mine 😅 .

My path to python.exe is `C:\\Users\\<my username>\\AppData\\Local\\lektor-cli\\lib\\Scripts\\python.exe` .
Thanks!

**Comment by xlotlu** [comment]
> My path to python.exe is `C:\\Users\\<my username>\\AppData\\Local\\lektor-cli\\lib\\Scripts\\python.exe` .

@fenekku right, that's the path to the python inside the virtualenv. How about outside of it?

**Comment by fenekku** [comment]
`C:\Users\<my username>\AppData\Local\Microsoft\WindowsApps\python.exe`

**Comment by xlotlu** [comment]
> `C:\Users\<my username>\AppData\Local\Microsoft\WindowsApps\python.exe`

@fenekku this means you're using Python from the Windows Store, and the lektor installer doesn't know how to handle that. Until we fix this, try using the official Python installer from https://www.python.org/downloads/windows/.

**Comment by runfalk** [comment]
@xlotlu do you think it's possible to detect this and display a more helpful error message?

**Comment by xlotlu** [comment]
> @xlotlu do you think it's possible to detect this and display a more helpful error message?

Well, anything is possible, it's just that someone has to do the work.

I don't remember all the details, but we discussed this in detail under https://github.com/lektor/lektor/pull/741#issuecomment-601724022. It's good we have an issue for it now.

If we go to the trouble of detecting it, we should just try to make it work. Until then, a documentation note would be welcome.

**Comment by fenekku** [comment]
Ok, after mucking around I got it to work! (clearly windows is not my main OS). Here is what worked for me:

1- install Python 3.8 via the windows installer: https://www.python.org/downloads/release/python-385/ and make sure to put python on the PATH.
2- open a cmd.exe / windows terminal with 'Run as administrator...'
3- In that terminal paste the command starting with `@powershell ...` (https://www.getlektor.com/docs/installation/)
4- Access lektor via `lektor` on the command line 

Thanks for the pointer to use the windows Python installation from the python.org site instead of the Windows app store.

This can be closed, although it would indeed be nice to document that installation subtlety.

**Comment by xlotlu** [comment]
> 2- open a cmd.exe / windows terminal with 'Run as administrator...'

It shouldn't be necessary to run things as admin.

> This can be closed

Let's keep it open, we should definitely support installing from windows store.

> ,although it would indeed be nice to document that installation subtlety.

Any pull request is welcome. The docs are momentarily part of the website repo: https://github.com/lektor/lektor-website/.
